R/view_landsat.R

```diff
@@ -0,0 +1,232 @@
+      )
+    )
+
+    false_color_tifs <- glue::glue_collapse(
+      tempfiles[imgs %in% c("B5.TIF", "B4.TIF", "B3.TIF")],
+      sep = " "
+    )
+
+    merge_rasts(out_file = false_file, in_files = false_color_tifs)
+
+    # build true color rast
+    true_file <- file.path(
+      tempdir(),
+      glue::glue(
+        "{scene$productId}_true_color.tif"
+      )
+    )
+
+    true_color_tifs <- glue::glue_collapse(
+      tempfiles[imgs %in% c("B4.TIF", "B3.TIF", "B2.TIF")],
+      sep = " "
+    )
+
+    merge_rasts(out_file = true_file, in_files = true_color_tifs)
+
+    # cirrus and QA bands
+    cirrus_band <- tempfiles[imgs == "B9.TIF"]
+    qa_band <- tempfiles[imgs == "BQA.TIF"]
+
+    cirrus_band_resamp <- resample_raster(
+      file = cirrus_band,
+      a_nodata = 0,
+      tr = 60
+    )
+    qa_band_resamp <- resample_raster(
+      file = qa_band,
+      a_nodata = 1,
+      tr = 60
+    )
+
+    system2(
+      "open",
+      c(cirrus_band_resamp, qa_band_resamp, false_file, true_file)
+    )
+
+
+    return("opening rasters in system viewer")
+  }
+
+  rast <- raster::stack(tempfiles)
+
+  raster::NAvalue(rast) <- 0
+
+  # fire up mapview
+  mapview::mapviewOptions(
+    raster.palette = viridisLite::viridis,
+    na.color = "#00000000"
+  )
+
+  # print false color map
+  composite <- mapview::viewRGB(
+    rast,
+    r = 1, g = 2, b = 3,
+    layer.name = "false color",
+    alpha.regions = 1
+  )
+
+  # print QA band
+  raster::NAvalue(rast[[6]]) <- 1
+  raster::values(rast[[6]]) <- factor(
+    as.character(raster::values(rast[[6]]))
+  )
+
+  qa <- mapview::mapview(
+    rast[[6]],
+    layer.name = "QA band"
+  )
+
+  # print cirrus band
+  cirrus <- mapview::mapview(
+    rast[[5]],
+    layer.name = "cirrus band",
+    alpha.regions = 1
+  )
+
+  # combine
+  qa + cirrus + composite
+
+}
+
+#' @title Generate table of Landsat scenes
+#'
+#' @return data.frame containing Landsat scene attributes from
+#' "https://landsat-pds.s3.amazonaws.com/c1/L8/scene_list.gz"
+#' @export
+#'
+
+get_scene_table <- function() {
+  scene_url <- "https://landsat-pds.s3.amazonaws.com/c1/L8/scene_list.gz"
+
+  scene_file <- file.path(
+    tempdir(),
+    basename(scene_url)
+  )
+
+  if (!file.exists(scene_file)) {
+    message("downloading list of scenes")
+    utils::download.file(
+      scene_url,
+      destfile = scene_file
+    )
+  }
+
+  out <-
+    vroom::vroom(
+      scene_file,
+      delim = ","
+    ) %>%
+    dplyr::mutate(
+      acquisitionDate = lubridate::as_date(acquisitionDate)
+    )
+
+  return(out)
+}
+
+
+#' @title Build multiband raster from several .tifs
+#'
+#' @param out_file file path to .tif file to be created
+#' @param in_files file paths of .tif files to be added to .vrt
+#'
+#' @return file path to output file
+
+merge_rasts <- function(out_file, in_files) {
+  merge_cmd <- glue::glue(
+    "gdal_merge.py -separate -n 0 -a_nodata 0 -ps 60 60 -co COMPRESS=LZW -o {out_file} {in_files}"
```

Comment:
does this imply the need for specific python libraries to be installed? Are those native? Should it check/error helpfully for users?

Reply:
gdal.py gets installed when you install the system package gdal. I don't think you need to install additional python packages but it's possible that you do and I did not notice.

Reply:
But will any random user necessarily have that installed?

Reply:
We should look at how they handle that for packages like sf - I think they have a section in readme about system dependencies like gdal